lib/puppet/resource/capability_finder.rb

```diff
@@ -22,6 +26,29 @@ def self.find(environment, code_id, cap)
       raise Puppet::DevError, 'PuppetDB is not available'
     end
 
+    resources = search(environment, nil, cap)
+
+    if resources.size > 1 && code_id
+      Puppet.debug "Found multiple resources when looking up capability #{cap}, filtering by code id #{code_id}"
+      resources = search(environment, code_id, cap)
+    end
+
+    if resources.size > 1
+      raise Puppet::DevError,
+        "Unexpected response from PuppetDB when looking up #{cap}:\n" +
+        "expected exactly one resource but got #{resources.size};\n" +
+        "returned data is:\n#{resources.inspect}"
+    end
+
+    if resource_hash = resources.first
+      instantiate_resource(resource_hash)
+    else
+      Puppet.debug "Could not find capability resource #{cap} in PuppetDB"
+      nil
```

Comment:
Is this nil really required here? Maybe pry is lying to me, but it seems like if you don't explicitly return nil, you will still get nil back when find is called

```
[1] pry(main)> def mymethod(test)
[1] pry(main)*   if test
[1] pry(main)*     puts "yay"
[1] pry(main)*   else
[1] pry(main)*     puts "no"
[1] pry(main)*     nil
[1] pry(main)*   end
[1] pry(main)* end
=> nil
[2] pry(main)> mymethod(false)
no
=> nil
[3] pry(main)> mymethod(true)
yay
=> nil
```

```
[5] pry(main)> def mymethod(test)
[5] pry(main)*   if test
[5] pry(main)*     puts "yay"
[5] pry(main)*   else
[5] pry(main)*     puts "no"
[5] pry(main)*   end
[5] pry(main)* end
=> nil
[6] pry(main)> mymethod(true)
yay
=> nil
[7] pry(main)> mymethod(false)
no
=> nil
```